linalg/petsc.cpp

```diff
@@ -1471,6 +1486,21 @@ void PetscParMatrix::Shift(const Vector & s)
    XX->ResetArray();
 }
 
+PetscParMatrix * TripleMatrixProduct(PetscParMatrix *R, PetscParMatrix *A,
+                                     PetscParMatrix *P)
+{
+   MFEM_VERIFY(A->Width() == P->Height(),
+               "Petsc TripleMatrixProduct: Number of local cols of A " << A->Width() <<
+               " differs from number of local rows of P " << P->Height());
+   MFEM_VERIFY(A->Height() == R->Width(),
+               "Petsc TripleMatrixProduct: Number of local rows of A " << A->Height() <<
+               " differs from number of local rows of R " << R->Width());
```

Comment:
```suggestion
               " differs from number of local cols of R " << R->Width());
```

@stefanozampini, can you make this and any other final changes?